core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -169,35 +180,109 @@ protected PlanNodeStatsEstimate visitLogicalExpression(LogicalExpression node, V
 
         private PlanNodeStatsEstimate estimateLogicalAnd(List<Expression> terms)
         {
-            // first try to estimate in the fair way
-            PlanNodeStatsEstimate estimate = process(terms.get(0));
-            if (!estimate.isOutputRowCountUnknown()) {
-                for (int i = 1; i < terms.size(); i++) {
-                    estimate = new FilterExpressionStatsCalculatingVisitor(estimate, session, types).process(terms.get(i));
+            List<PlanNodeStatsEstimate> estimates = estimateCorrelatedExpressions(terms);
+            // Sort known estimates by selectivity
+            List<PlanNodeStatsEstimate> knownSortedEstimates = estimates.stream()
+                    .filter(estimate -> !estimate.isOutputRowCountUnknown())
+                    .sorted(Comparator.comparingDouble(PlanNodeStatsEstimate::getOutputRowCount))
+                    .collect(toImmutableList());
+
+            if (knownSortedEstimates.isEmpty()) {
+                return PlanNodeStatsEstimate.unknown();
+            }
+
+            double combinedSelectivity = knownSortedEstimates.get(0).getOutputRowCount() / input.getOutputRowCount();
+            double combinedIndependenceFactor = 1.0;
+            double independenceFactor = getFilterConjunctionIndependenceFactor(session);
+            // For independenceFactor = 0.75 and terms t1, t2, t3
+            // Combined selectivity = (t1 selectivity) * ((t2 selectivity) ^ 0.75) * ((t3 selectivity) ^ (0.75 * 0.75))
+            // independenceFactor = 1 implies the terms are assumed to have no correlation and their selectivities are multiplied without scaling.
+            // independenceFactor = 0 implies the terms are assumed to be fully correlated and only the most selective term drives the selectivity.
+            for (int i = 1; i < knownSortedEstimates.size(); i++) {
```

Comment:
Since this is a heuristic, it would be useful to have plan tests that showcase both the best and the worst case for this. Later on, we could use these tests to fix the worst case without regression on the best + it would help in understanding how this heuristic affects query plans,

Reply:
TPC benchmarks contain both perfectly correlated and un-correlated cases. For any changes here we would rely on AbstractCostBasedPlanTest to identify changes to query plans and run the benchmarks to find out the impact.